workflow/templateresolution/context.go

```diff
@@ -281,9 +283,16 @@ func (ctx *Context) WithWorkflowTemplate(name string) (*Context, error) {
 
 // WithWorkflowTemplate creates new context with a wfv1.TemplateHolder.
 func (ctx *Context) WithClusterWorkflowTemplate(name string) (*Context, error) {
-	cwftmpl, err := ctx.cwftmplGetter.Get(name)
+	cwftmpl, err := ctx.GetClusterWorkflowTemplate(name)
 	if err != nil {
 		return nil, err
 	}
 	return ctx.WithTemplateBase(cwftmpl), nil
 }
+
+func (ctx *Context) GetClusterWorkflowTemplate(name string) (*wfv1.ClusterWorkflowTemplate, error) {
+	if ctx.cwftmplGetter == nil && reflect.ValueOf(ctx.cwftmplGetter).IsNil() {
```

Comment:
why do you use reflect here? typically reflection is expensive and should be avoided - could you replace cwtmplGetter using the null object pattern? like what we do for the workflow archive and offload node status?

Reply:
Got it. I like the Null Object Pattern